contracts/mirrorrules.sol

```diff
@@ -0,0 +1,90 @@
+pragma solidity ^0.4.0;
+import "./abstracts/trialrulesabstract.sol";
+
+
+contract MirrorRules is TrialRulesAbstract {
+
+
+    uint8 constant MIRROR_CHALLENGE = uint8(TrialRulesAbstract.Status.NOT_GUILTY) + 1;
+    uint8 constant VALID_MIRROR_CHALLENGE = uint8(TrialRulesAbstract.Status.NOT_GUILTY) + 2;
+    uint constant MIRROR_CHALLENGE_GRACE_PERIOD = 35;//Grace period set to 35 blocks to submit evidence.
+    uint constant VALID_MIRROR_CHALLENGE_GRACE_PERIOD = 35;
+    uint constant REWARD = 5; //plaintiff reward for the case of a valid case
+
+    mapping(uint8 => address) public witnesses;
+    mapping(uint8 => mapping(uint8 => uint8)) public transitions;
+    mapping(uint8 => uint) public gracePeriods;
+    //map caseId to map status to time elapse
+    mapping(bytes32 => mapping(uint8 => uint)) gracePeriodStartTime;
+
+    function MirrorRules(address paymentValidatorContract,address ensMirrotValidatorContract) {
```

Comment:
ensMirrotValidatorContract should be ensMirrorValidatorContract, same with use of variable on line 23